tools/coverity/misra.config

```diff
@@ -0,0 +1,42 @@
+// MISRA C-2012 Rules
+
+{
+    version : "2.0",
+    standard : "c2012",
+    title: "Coverity MISRA Configuration",
+    deviations : [
+        // Disable the following rules.
+        {
+            deviation: "Directive 4.5",
+            reason: "Allow names that MISRA considers ambiguous."
+        },
+        {
+            deviation: "Directive 4.8",
+            reason: "Allow inclusion of unused types. Header files for coreMQTT, which are needed by all files, define types that are not used by the agent."
```

Comment:
```suggestion
            reason: "Allow inclusion of unused types. Header files for coreMQTT-Agent, which are needed by all files, define types that are not used by the agent."
```

Reply:
Please handle that for all the similar instances

Reply:
No, this is correct. The MISRA warnings from removing this from the config are due to coreMQTT header files.

Reply:
Yea. Sounds good. Thanks